jenkins/rootfs-builder.jpl

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env groovy
+
+/*
+  Copyright (C) 2020 Collabora Limited
+  Author: Lakshmipathi Ganapathi <lakshmipathi.ganapathi@collabora.com>
+
+  This module is free software; you can redistribute it and/or modify it under
+  the terms of the GNU Lesser General Public License as published by the Free
+  Software Foundation; either version 2.1 of the License, or (at your option)
+  any later version.
+
+  This library is distributed in the hope that it will be useful, but WITHOUT
+  ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+  FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+  details.
+
+  You should have received a copy of the GNU Lesser General Public License
+  along with this library; if not, write to the Free Software Foundation, Inc.,
+  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+*/
+
+/* ----------------------------------------------------------------------------
+ * Jenkins parameters
+
+KCI_CORE_URL (https://github.com/kernelci/kernelci-core.git)
+  URL of the kernelci-core repository
+KCI_CORE_BRANCH (master)
+  Name of the branch to use in the kernelci-core repository
+KCI_API_URL (https://api.kernelci.org)
+  URL of the KernelCI backend API
+KCI_TOKEN_ID
+  Identifier of the KernelCI backend API token stored in Jenkins
+DOCKER_BASE
+  Dockerhub base address used for the build images
+ROOTFS_CONFIG
+  Configuration name to build RootFS images
+ROOTFS_ARCH
+  RootFS image architecture
+*/
+
+
+@Library('kernelci') _
+import org.kernelci.util.Job
+
+def build_rootfs(config, arch, pipeline_version,kci_core) {
+
+    dir(kci_core) {
+        sh(script: """\
+./kci_rootfs \
+build \
+--config ${config} \
+--arch  ${arch} \
+--data-path jenkins/debian/debos 
+mkdir -p ${pipeline_version} 
+mv jenkins/debian/debos/${config}/* ${pipeline_version}
+""")
+
+    archiveArtifacts artifacts: "${pipeline_version}/**"
+
+    withCredentials([string(credentialsId: params.KCI_TOKEN_ID, variable: 'API_TOKEN')]) {
+            sh(script: """\
+./kci_rootfs \
+upload \
+--token ${API_TOKEN} \
+--api ${params.KCI_API_URL} \
+--rootfs-dir ${pipeline_version} \
+--upload-path images/rootfs/debian/${config}/${pipeline_version}
+""")
+
+        }
+    }
+
+}
+
+node("debos && docker") {
+    def j = new Job()
+    def kci_core = "${env.WORKSPACE}/kernelci-core"
+    def config = "${params.ROOTFS_CONFIG}"
+    def arch = "${params.ROOTFS_ARCH}"
+    def docker_image = "${params.DOCKER_BASE}debos"
+    def pipeline_version =  "${params.PIPELINE_VERSION}"
+
+    j.dockerPullWithRetry(docker_image).inside() {
+        j.cloneKciCore(kci_core, params.KCI_CORE_URL, params.KCI_CORE_BRANCH)
+    }
+
+    j.dockerPullWithRetry(docker_image).inside(" --privileged --device /dev/kvm ") {
+        stage("Build ${config} ${arch}") {
+            timeout(time: 60, unit: 'MINUTES') {
```

Comment:
Please increase the timeout to 4h i.e. 240 minutes.  Some test suites may take a very long time to build.